final_task/rss_reader/databaseEmulation.py

```diff
@@ -0,0 +1,94 @@
+import pickle
+import logging as log
+import os
+import datetime
+
+from typing import Any
+
+
+now = datetime.datetime.now
+
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(
+                *args,
+                **kwargs
+            )
+        return cls._instances[cls]
+
+
+class DatabaseEmulation(metaclass=Singleton):
+    def __init__(self, path_to_dates, path_to_data):
+        self.path_to_dates = path_to_dates
+        self.path_to_data = path_to_data
+        self.dates = {}
+        with open(path_to_dates, "r") as data_file:
+            for line in data_file:
+                if not line:
+                    continue
+                date, row_number = line.strip().split()
+                self.dates[date] = int(row_number)
+        self.dates_file = open(path_to_dates, "a")
+        self.write_data_stream = open(self.path_to_data, "ab")
+        self.read_data_stream = open(self.path_to_data, "rb")
+
+    def __del__(self):
+        self.dates_file.close()
+        self.write_data_stream.close()
+        self.read_data_stream.close()
```

Comment:
It is better to use context manager's `__enter__` and `__exit__` for this purpose, not `__del__`